src/config/normalize.js

```diff
@@ -1,12 +1,14 @@
-import { Dimensions } from 'react-native';
+import { Dimensions, PixelRatio } from 'react-native';
 
-// Default guideline sizes are based on standard ~6.12" screen mobile device (iPhone 14 Pro)
 const guidelineBaseWidth = 393;
+const guidelineBaseHeight = 852;
 
 export const normalize = (size, factor = 1) => {
-  const { width, height } = Dimensions.get('window');
-  const shortDimension = width < height ? width : height;
-  const scale = (shortDimension / guidelineBaseWidth) * size;
+   const { width, height } = Dimensions.get('window');
+   const widthScale = width / guidelineBaseWidth;
+   const heightScale = height / guidelineBaseHeight;
+   const scale = Math.min(widthScale, heightScale); // Takes the smaller ratio to avoid excess.
```

Comment:
In which cases would the `widthScale` ne smaller and when the `heightScale`?

Reply:
`widthScale` is smaller when the current device is narrower and longer.
`heightScale` is smaller when the current device is wider and shorter.

**Example with actual numbers:**

**Example 1:** Device with 360px width and 640px height
- `widthScale = 360 / 393 ≈ 0.915`
- `heightScale = 640 / 852 ≈ 0.751`
- Here, `heightScale` is smaller than `widthScale` because the device is wider and shorter relative to the baseline size.

**Example 2:** Device with 400px width and 800px height
- `widthScale = 400 / 393 ≈ 1.018`
- `heightScale = 800 / 852 ≈ 0.939`
- Here, `widthScale` is smaller than `heightScale` because the device is narrower and longer relative to the baseline size.

This logic ensures that the scaling works well for different devices by always using the smaller scale to maintain proportional UI elements, preventing them from being too large or too small.

Reply:
In example 2 `heightScale` is also smaller with 0.939 compared to 1.018. Was ChatGPT going wrong here ;)?

Reply:
well yeah :)...
but we get the idea, don't we? that in some cases the `heightscale` gets nearer and also might be bigger than `widthScale`. 
for example, with some tablets or tablet like phones.

Reply:
i really don't get it exactly. can you name one specific device width its sizes that we can run a simulator with to see how things look?

Reply:
for example a landscape tablet so we get the smaller scale to make all components appear smaller adjusted to the device.